react/index.js

```diff
@@ -33,8 +39,97 @@ export class MiniCart extends Component {
 
   state = {
     openContent: false,
+    updatingOrderForm: false,
   }
 
+  debounce = null
+
+  async componentDidUpdate(prevProps, prevState) {
+    const serverItems = path(['data', 'orderForm', 'items'], this.props)
+    const clientItems = path(['linkState', 'minicartItems'], this.props)
+
+    if (serverItems && clientItems) {
+      const prevClientItems = path(['linkState', 'minicartItems'], prevProps)
+
+      if (!equals(prevClientItems, clientItems)) {
+        if (this.debounce) {
+          clearTimeout(this.debounce)
+        }
+        this.debounce = setTimeout(
+          () => this.handleItemsDifference({ clientItems }),
+          2e3
+        )
+      } else if (serverItems.length && !clientItems.length) {
+        return this.fillClientMinicart(serverItems)
+      } else if (prevState.updatingOrderForm && !this.state.updatingOrderForm) {
+        const {
+          data: {
+            orderForm: { items },
+          },
+        } = await this.props.data.refetch()
+        return this.fillClientMinicart(items)
+      }
+    }
+  }
+
+  handleItemsDifference = async ({ clientItems }) => {
+    const items = clientItems
+      .map(({ id, index, quantity, seller }) => ({
+        id,
+        index,
+        quantity,
+        seller,
+      }))
+      .filter(
+        ({ id, index, quantity, seller }) =>
+          !isNil(id) && !isNil(index) && !isNil(quantity) && !isNil(seller)
```

Comment:
Do you need that filter? Isn't this data trustable?

Reply:
No, it isn't.